components/ReplyItem.js

```diff
@@ -1,55 +1,426 @@
+  mutation CreateOrUpdateArticleReplyFeedback(
+    $articleId: String!
+    $replyId: String!
+    $vote: FeedbackVote!
+    $comment: String
+  ) {
+    CreateOrUpdateArticleReplyFeedback(
+      articleId: $articleId
+      replyId: $replyId
+      vote: $vote
+      comment: $comment
+    ) {
+      ...ArticleReplyFeedbackData
+    }
+  }
+  ${ArticleReplyFeedbackData}
+`;
 /**
  *
  * @param {ReplyItem} props.reply - see ReplyItem in GraphQL fragment
  * @param {boolean} showUser
  */
-function ReplyItem({ reply, showUser = true }) {
-  const { type: replyType } = reply;
+function ReplyItem({
+  articleId,
+  replyId,
+  reply,
+  feedbacks,
+  positiveFeedbackCount,
+  negativeFeedbackCount,
+}) {
+  const [createReplyFeedback] = useMutation(CREATE_REPLY_FEEDBACK, {
+    onCompleted() {
+      closeVotePopover();
+    },
+  });
+
+  const [vote, setVote] = useState(null);
+  const [reason, setReason] = useState('');
+  const [reasonsPopoverAnchorEl, setReasonsPopoverAnchorEl] = useState(null);
+  const [votePopoverAnchorEl, setVotePopoverAnchorEl] = useState(null);
+  const [tab, setTab] = useState(0);
+  const { user, text, type: replyType } = reply;
   const createdAt = new Date(reply.createdAt);
   const timeAgoStr = formatDistanceToNow(createdAt);
 
+  const openReasonsPopover = event => {
+    setReasonsPopoverAnchorEl(event.currentTarget);
+  };
+
+  const closeReasonsPopover = () => {
+    setReasonsPopoverAnchorEl(null);
+  };
+
+  const openVotePopover = (event, value) => {
+    setVotePopoverAnchorEl(event.currentTarget);
+    setVote(value);
+  };
+
+  const closeVotePopover = () => {
+    setVotePopoverAnchorEl(null);
+    setVote(null);
+  };
+
+  const classes = useStyles({ replyType });
+
   return (
-    <Link href="/reply/[id]" as={`/reply/${reply.id}`}>
-      <a className="item">
-        <div title={TYPE_NAME[replyType]}>{TYPE_ICON[replyType]}</div>
-        <div className="item-content">
-          <div className="item-text">
-            {showUser ? `${reply?.user?.name || '有人'}：` : ''}
-            {reply.text}
-          </div>
-          <div className="item-info">
-            使用於 {reply.articleReplies.length} 篇
-            {isValid(createdAt) ? (
-              <span title={format(createdAt)}>・{t`${timeAgoStr} ago`}</span>
-            ) : (
-              ''
-            )}
+    <div className={classes.root}>
+      <Box p="24px" display={['none', 'none', 'block']}>
+        <Avatar user={user} size={72} />
+        {/*
+          <div title={TYPE_NAME[replyType]}>{TYPE_ICON[replyType]}</div>
+        */}
+      </Box>
+      <Box pr="14px" display={['block', 'block', 'none']}>
+        <Avatar user={user} size={30} />
+        {/*
+          <div title={TYPE_NAME[replyType]}>{TYPE_ICON[replyType]}</div>
+        */}
+      </Box>
+      <Box py="12px" flexGrow={1}>
+        <div
+          className={classes.replyType}
+        >{t`${user.name} consider this ${TYPE_NAME[replyType]}`}</div>
```

Comment:
Another thing is, I am wondering if we should use `reply`'s user name, or `articleReply`'s user name, since they can be different users, and the original author of the reply could think differently from the person that connects their reply to some other messages.

Discussion opened here: https://www.figma.com/file/zpD45j8nqDB2XfA6m2QskO?node-id=79:9#23909174

Reply:
about reply's user name I'll take a note and remain the same for the time being, because I haven't figured out the "reference another reply" logic yet

Reply:
Got it, let's save this to future PRs